src/localplayer.cpp

```diff
@@ -309,17 +358,28 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 
 			// The node to be sneaked on has to be walkable
 			node = map->getNodeNoEx(p, &is_valid_position);
-			if (!is_valid_position || nodemgr->get(node).walkable == false)
-				continue;
-			// And the node above it has to be nonwalkable
-			node = map->getNodeNoEx(p + v3s16(0,1,0), &is_valid_position);
-			if (!is_valid_position || nodemgr->get(node).walkable)
+			if (!is_valid_position || !nodemgr->get(node).walkable)
 				continue;
+			// And the node(s) above have to be nonwalkable
+			bool ok = true;
 			if (!physics_override_sneak_glitch) {
```

Comment:
Perhaps because it's good to have the default / primary code first?
But if sfan5 doesn't mind i would prefer what rubenwardy requests.